ee/audit/run_ee.go

```diff
@@ -107,37 +107,116 @@ func run() error {
 		glog.Info("audit file is empty")
 		return nil
 	}
-	var iterator int64 = 0
 
-	iv := make([]byte, aes.BlockSize)
-	x.Check2(file.ReadAt(iv, iterator))
-	iterator = iterator + aes.BlockSize
+	// decrypt header in audit log to verify encryption key
+	// [16]byte IV + [4]byte len(x.VerificationText) + [11]byte x.VerificationText
+	decryptHeader := func() ([]byte, int64, error) {
+		var iterator int64 = 0
+		iv := make([]byte, aes.BlockSize)
+		x.Check2(file.ReadAt(iv, iterator))     // get first iv
+		iterator = iterator + aes.BlockSize + 4 // length of verification text encoded in uint32
 
-	t := make([]byte, len(x.VerificationText))
-	x.Check2(file.ReadAt(t, iterator))
-	iterator = iterator + int64(len(x.VerificationText))
+		t := make([]byte, len(x.VerificationText))
+		x.Check2(file.ReadAt(t, iterator))
+		iterator = iterator + int64(len(x.VerificationText))
 
-	stream := cipher.NewCTR(block, iv)
-	stream.XORKeyStream(t, t)
-	if string(t) != x.VerificationText {
-		return errors.New("invalid encryption key provided. Please check your encryption key")
+		text := make([]byte, len(x.VerificationText))
+		stream := cipher.NewCTR(block, iv)
+		stream.XORKeyStream(text, t)
+		if string(text) != x.VerificationText {
+			return nil, 0, errors.New("invalid encryption key provided. Please check your encryption key")
+		}
+		return iv, iterator, nil
 	}
 
-	for {
-		// if its the end of data. finish decrypting
-		if iterator >= stat.Size() {
-			break
-		}
-		x.Check2(file.ReadAt(iv[12:], iterator))
-		iterator = iterator + 4
+	// [12]byte baseIV + [4]byte len(x.VerificationText_Deprecated) + [11]byte x.VerificationText_Deprecated
+	decryptHeader_Deprecated := func() ([]byte, int64, error) {
+		var iterator int64 = 0
+		iv := make([]byte, aes.BlockSize)
+		x.Check2(file.ReadAt(iv, iterator))
```

Comment:
does it make sense to crash when we can't read from the file?

Reply:
Good catch...  this is indeed an issue because we may try using multiple decrypt methods and don't want to crash and should handle the error more gracefully.